tasks/new_e2e_tests.py

```diff
@@ -148,44 +148,40 @@ def _clean_locks():
     for entry in os.listdir(Path(lock_dir)):
         subdir = os.path.join(lock_dir, entry)
         for filename in os.listdir(Path(subdir)):
-            file_path = os.path.join(subdir, filename)
-            if os.path.isfile(file_path) and filename.endswith(".json"):
-                os.remove(file_path)
-                print(f"🗑️ Deleted lock: {file_path}")
+            path = os.path.join(subdir, filename)
+            if os.path.isfile(path) and filename.endswith(".json"):
+                os.remove(path)
+                print(f"🗑️ Deleted lock: {path}")
+            elif os.path.isdir(path):
+                shutil.rmtree(path)
 
 
-def _clean_stacks():
+def _clean_stacks(ctx: Context):
     print("🧹 Clean up stacks")
-    stacks = _get_existing_stacks()
+    stacks = _get_existing_stacks(ctx)
 
     for stack in stacks:
         print(f"🗑️ Cleaning up stack {stack}")
-        _remove_stack(stack)
+        _remove_stack(ctx, stack)
 
 
-def _get_existing_stacks() -> List[str]:
+def _get_existing_stacks(ctx: Context) -> List[str]:
     # ensure we deal with local stacks
-    output = subprocess.check_output(["pulumi", "stack", "ls", "--all"], cwd=tempfile.gettempdir())
-    output = output.decode("utf-8")
-    lines = output.splitlines()
-    lines = lines[1:]  # skip headers
-    e2e_stacks: List[str] = []
-    for line in lines:
-        stack_name = line.split(" ")[0]
-        e2e_stacks.append(stack_name)
-    return e2e_stacks
-
-
-def _remove_stack(stack_name: str):
-    subprocess.call(
-        [
-            "pulumi",
-            "stack",
-            "rm",
-            "--force",
-            stack_name,
-        ]
-    )
+    with ctx.cd('/tmp'):
```

Comment:
Why not using `tempfile.gettempdir()`?

Reply:
Actually better yes, compatible with Windows too. Changing this.